htdocs/product/stock/productlot_card.php

```diff
@@ -596,7 +596,7 @@
 	// Documents
 	if ($includedocgeneration) {
 		$objref = dol_sanitizeFileName($object->ref);
-		$relativepath = $objref.'/'.$objref.'.pdf';
+		$relativepath = $object->id.'/'.$objref.'.pdf';
```

Comment:
Using id may be a problem when doing export/import
Id may change. Restoring directory of a backup for exemple 
will be  not sync with restore 
 This is why we use ref.
Possible to use productref_batchnumber instead of batch number ? (I think it is the real key of a batch number)

Possible to use